### DISCUSSION TITLE
Is this good convention?

### DISCUSSION THREAD

**Discussion by JSchugt** [nss-day-cohort-47/C47-Discussions#39] [Discuss a topic]
After getting my NASA keys (hopefully it's to Ingenuity)  that URL is HUGE.    

so I saved the URL   in a variable
`const base_url = `https://api.nasa.gov/planetary/apod?api_key=${apodKey}`;`  

apodKey is a variable that is being imported from a file that has been added to the .gitignore. 

Is this bad form, or start of good practice?

**Comment by JSchugt** [comment]
code snippet 


'// URL for APOD Key

import {apodKey} from "./keys/keys.js";
const base_url = `https://api.nasa.gov/planetary/apod?api_key=${apodKey}`;
let apodResposne = [];
export const getAPOD = () => {
    return fetch(base_url, {
        method: "GET",
        headers: {
            "Content-Type": "application/json"
        }
    })
    .then(respone => respone.json())
    .then(paresedResponse => apodResposne = paresedResponse)
}

export const useApod = () => {
    return apodResposne;
}
'

**Reply by BryanNilsen** [reply]
This is great! I definitely think you're on the right track and the start to a good practice.
You don't want to expose your api key on github, so storing it in a file and gitignoring it is a good way to go. You've stumbled across environmental variables! Storing the base url in a variable is also good if you've got multiple requests so you don't have to retype `https://api.nasa.gov/planetary/apod?api_key=${apodKey}` for each request.

I'll just nitpick the typos... there's a bunch of 'em, so I'll put them in an array
const typos = [ "paresedResponse", "apodResponsne", "respone"] 🤓

**Reply by pour-theoil** [reply]
Does that mean you would put the keys.js in the .gitignore file?

**Reply by BryanNilsen** [reply]
☝️ yes. HOWEVER... understand though that on the client side, when making requests, ultimately you expose your api key to anyone who cares to look. What this does is only keep the key out of the repository on GitHub... which is fine for now.
